src/dataExplorer/context/scripts.tsx

```diff
@@ -0,0 +1,97 @@
+import React, {
+  FC,
+  createContext,
+  useContext,
+  useState,
+  useCallback,
+} from 'react'
+import {useDispatch} from 'react-redux'
+import * as api from 'src/client/scriptsRoutes'
+import {notify} from 'src/shared/actions/notifications'
+import {
+  scriptSaveFail,
+  scriptSaveSuccess,
+} from 'src/shared/copy/notifications/categories/scripts'
+import {PersistanceContext} from 'src/dataExplorer/context/persistance'
+
+interface ContextType {
+  description: string
+  name: string
+
+  handleSave: () => void
+  updateDescription: (description: string) => void
+  updateName: (name: string) => void
+}
+
+const DEFAULT_CONTEXT = {
+  description: '',
+  name: `Untitled Script: ${new Date().toISOString()}`,
+
+  handleSave: () => {},
+  updateDescription: (_description: string) => {},
+  updateName: (_name: string) => {},
+}
+
+export const ScriptContext = createContext<ContextType>(DEFAULT_CONTEXT)
+
+export const ScriptProvider: FC = ({children}) => {
```

Comment:
i figure imma be replacing most of this, but before i blindly do so, what is important to keep in this provider?